eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/tasklets/buildReport/BuildReportTasklet.java

```diff
@@ -0,0 +1,122 @@
+
+/**
+ * Reads the 2 temporary files with contigs and variants written by {@link AccessionReportWriter} and writes a
+ * complete VCF.
+ *
+ * This extra step is needed because the writer can not deal with resumed jobs. It could write those files and
+ * concatenate them when the "close" method was called, but it is impossible to differentiate if it is being closed
+ * because of an error in the batch or because all elements are finished. The temporary files should be kept in the
+ * first case and deleted in the second case, so this extra step is the place to write the final VCF report and delete
+ * the temporary files.
+ */
+public class BuildReportTasklet implements Tasklet {
+
+    private static final Logger logger = LoggerFactory.getLogger(BuildReportTasklet.class);
+
+    private final File contigsFile;
+
+    private final File variantsFile;
+
+    private File output;
+
+    public BuildReportTasklet(File output) {
+        this.output = output;
+        this.contigsFile = new File(output.getAbsolutePath() + CONTIGS_FILE_SUFFIX);
+        this.variantsFile = new File(output.getAbsolutePath() + VARIANTS_FILE_SUFFIX);
+    }
+
+    @Override
+    public RepeatStatus execute(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(output));
+        writeHeader(writer, getContigPairs(contigsFile));
+        appendFileToWriter(writer, variantsFile);
+        writer.close();
+        variantsFile.delete();
+        contigsFile.delete();
+
+        return RepeatStatus.FINISHED;
+    }
+
+    private Stream<Pair<String, String>> getContigPairs(File contigsFile) throws IOException {
+        return Files.lines(contigsFile.toPath()).map(line -> {
+            String[] contigColumns = line.split("\t");
+            if (contigColumns.length != 2) {
+                throw new IllegalStateException("Temporary file " + contigsFile.getAbsolutePath()
+                                                + " doesn't have the expected format. Please delete it and "
+                                                + "start a new job.");
+            }
+            String originalContig = contigColumns[0];
+            String insdcContig = contigColumns[1];
+            return Pair.of(originalContig, insdcContig);
+        });
+    }
+
+    private void writeHeader(BufferedWriter writer, Stream<Pair<String, String>> inputContigAndInsdcPairs)
+            throws IOException {
+        writer.write("##fileformat=VCFv4.2");
+        writer.newLine();
+
+        inputContigAndInsdcPairs.forEach((Pair<String, String> inputContigAndInsdc) -> {
+            try {
+                writer.write("##contig=<ID=" + inputContigAndInsdc.getFirst() + ",Description=\""
+                             + inputContigAndInsdc.getSecond() + "\">");
+                writer.newLine();
+            } catch (IOException e) {
```

Comment:
Why do we handle an exception here but not in the method below? Also why should it be transformed to an RTE?

Reply:
lambdas can throw exceptions in java, but classes like Function can't do so:
```
@FunctionalInterface
public interface Function<T, R> {
    R apply(T t);    // <-- no checked exceptions in signature!
    [...]
```